django/db/models/fields/reverse_related.py

```diff
@@ -115,6 +115,23 @@ def __repr__(self):
             self.related_model._meta.model_name,
         )
 
+    def __eq__(self, other):
+        return (
+            isinstance(other, self.__class__) and
+            self.field == other.field and
+            self.model == other.model and
+            self.related_name == other.related_name and
+            self.related_query_name == other.related_query_name and
+            self.limit_choices_to == other.limit_choices_to and
+            self.parent_link == other.parent_link and
+            self.on_delete == other.on_delete and
+            self.symmetrical == other.symmetrical and
+            self.multiple == other.multiple
+        )
+
+    def __hash__(self):
+        return super().__hash__()
+
```

Comment:
Looks like Github lost my comment somehow, not sure how.

Per [Python's documentation about `__hash__`](https://docs.python.org/3/reference/datamodel.html#object.__hash__).

> The only required property is that objects which compare equal have the same hash value; it is advised to mix together the hash values of the components of the object that also play a part in comparison of objects by packing them into a tuple and hashing the tuple.

Therefore I suggest we define a property that returns a tuple reusable by both methods.

```suggestion
    @property
    def identity(self):
        return (
            self.field,
            self.model,
            self.related_name,
            self.related_query_name,
            tuple(sorted(self.limit_choices_to.items())) if isinstance(self.limit_choices_to, dict) else self.limit_choices_to,
            self.parent_link,
            self.on_delete,
            self.symmetrical,
            self.multiple,
        ) 

    def __eq__(self, other):
        return isinstance(other, self.__class__) and other.identity == self.identity

    def __hash__(self):
        return hash(self.identity)

```